DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+```java
+  @AfterEach
+  void destroy() throws Exception {
+    reset(drive);
+  }
+```
+
+Once everything is set up, you can make tests with the `@Test` annotation. Here's a quick example of testing if our motors are getting voltage:
+```java
+   @Test
+  public void testSparks() {
+    drive.drive(1, 0);
+
+    assert spark.getBusVoltage() > 3;
+  }
+  ```
+  - 3 is a random number here, but it is small on purpose. The motor will need more voltage than 3 in order to properly run.
+
+  Another example: Here we are testing to see if our pose is changing as we drive: 
+```java
+  @Test
+  public void testPose(){
+    drive.drive(0.8, 0.8);
+    fastForward();
+
+    assertNotEquals(new Pose2d(), drive.getPose());
+  }
+```
+  #### Running the tests
+To actually see if the tests are valid, open the wpilib command palette(ctrl + shift + p) and search ```Test Robot Code``` and press enter. After a bit, it will show you the results of all your tests in the terminal. 
+
+![alt text](image-3.png)
+
+Unit tests should be made for most of your subsystems and are a great way to see if the logic behind the code actually makes sense and works. You should also try to test all parts of the subsystems to cover everything. 
+
+Here's another simple test but for LEDs. 
+```
+@Test
+  public void testRainbowTheme() {
+    var rainbow1 = LedStrip.getBufferDataString(led.rainbowAddressableLEDBuffer());
+    fastForward(2);
+    var rainbow2 = LedStrip.getBufferDataString(led.rainbowAddressableLEDBuffer());
+    assertNotEquals(rainbow1, rainbow2);
+  }
+  ```
+  If you just scan through the code, its easy to tell what the process is and what we're testing. Setting up the leds to a pattern, forwarding in time so that its a different color than before, and testing to make sure that the two colors are indeed different from each other. 
+
+  The last type of assert most commonly used in tests is the `assertEqual`. This simply tests if two values are the same. Both in assertEqual and assertNotEqual, you can pass in a tolerance value that will pass the test if the two different values are within the tolarence. 
+  
+#### System Checks
+In addition to unit tests, it’s equally important to perform system checks to ensure the entire drivetrain is operating as expected in real-world conditions. Think of system checks as a quick health check for your robot before it hits the field. You want to make sure that everything is working as it should—motors, sensors, and all. This is usually done in real life and it's a good habit to run full robot system checks before your matches. 
+
+Here is a quick simple example of what a part of a drive system check might look like. The logic is similar to unit tests as we will run a part of the drive, and then check to see if it worked. Remember to import the Sciborg's Test Library and not the wpilib built in one. 
+
+Step 1: Define the command to run the system check
```

Comment:
Note that this is created in the subsystem b/c of easy access to internal methods to check conditions.